--- original_tracetreeitem.cpp	2022-06-27 00:34:55.075529700 +0200
+++ tracetreeitem.cpp	2022-06-27 00:28:52.480286600 +0200
@@ -18,6 +18,7 @@
  */
 
 #include <cassert>
+#include <QMenu>
 
 #include "view.hpp"
 
@@ -138,6 +139,43 @@
 	return QPoint(rect.right(), get_visual_y());
 }
 
+QMenu* TraceTreeItem::create_view_context_menu(QWidget *parent, QPoint &click_pos)
+{
+    (void)click_pos;
+
+    QMenu *const menu = new QMenu(parent);
+
+    View *view = owner_->view();
+
+    if(view->ruler_offset().convert_to<double>() != 0){
+        QAction *const reset_view = new QAction(tr("Reset view"), this);
+        connect(reset_view, SIGNAL(triggered()), this, SLOT(on_view_reset()));
+        menu->addAction(reset_view);
+    }
+
+    if(view->scale() != 1e-3) {
+        QAction *const reset_zoom = new QAction(tr("Reset zoom"), this);
+        connect(reset_zoom, SIGNAL(triggered()), this, SLOT(on_zoom_reset()));
+        menu->addAction(reset_zoom);
+    }
+
+    return menu;
+}
+
+void TraceTreeItem::on_view_reset()
+{
+    View *view = owner_->view();
+    assert(view);
+    view->reset_offset();
+}
+
+void TraceTreeItem::on_zoom_reset()
+{
+    View *view = owner_->view();
+    assert(view);
+    view->reset_zoom();
+}
+
 } // namespace trace
 } // namespace views
 } // namespace pv
